code/datums/status_effects/buffs/stun_absorption.dm

```diff
@@ -0,0 +1,236 @@
+	if(amount > 0 SECONDS)
+		// Show the message
+		if(shown_message)
+			// We do this replacement meme, instead of just setting it up in creation,
+			// so that we respect indentity changes done while active
+			var/really_shown_message = replacetext(shown_message, "%EFFECT_OWNER", "[owner]")
+			owner.visible_message(really_shown_message, ignored_mobs = owner)
+
+		// Send the self message
+		if(self_message)
+			to_chat(owner, self_message)
+
+		// Count seconds absorbed
+		seconds_of_stuns_absorbed += amount
+		if(seconds_of_stuns_absorbed >= max_seconds_of_stuns_blocked)
+			qdel(src)
+
+	return .
+
+/**
+ * [proc/apply_status_effect] wrapper specifically for [/datum/status_effect/stun_absorption],
+ * specifically so that it's easier to apply stun absorptions with named arguments.
+ *
+ * If the mob already has a stun absorption from the same source, will not re-apply the effect,
+ * unless the new effect's priority is higher than the old effect's priority.
+ *
+ * Arguments
+ * * source - the source of the stun absorption.
+ * * duration - how long does the stun absorption last before it ends? -1 or null = infinite duration
+ * * priority - what is this effect's priority to other stun absorptions? higher = more priority
+ * * message - optional, "other message" arg of visible message, shown on trigger. Use %EFFECT_OWNER if you want the owner's name to be inserted.
+ * * self_message - optional, "self message" arg of visible message, shown on trigger
+ * * examine_message - optional, what is shown on examine of the mob.
+ * * max_seconds_of_stuns_blocked - optional, how many seconds of stuns can it block before deleting? the stun that breaks over this number is still blocked, even if it is much higher.
+ *
+ * Returns an instance of a stun absorption effect, or NULL if failure
+ */
+/mob/living/proc/add_stun_absorption(
+	source,
+	duration,
+	priority = -1,
+	message,
+	self_message,
+	examine_message,
+	max_seconds_of_stuns_blocked = INFINITY,
+)
+
+	// Handle duplicate sources
+	for(var/datum/status_effect/stun_absorption/existing_effect in status_effects)
+		if(existing_effect.source != source)
+			continue
+
+		// If an existing effect's priority is greater or equal to our passed priority...
+		if(existing_effect.priority >= priority)
+			// don't bother re-applying the effect, and return
+			return
+
+		// otherwise, delete existing and replace with new
+		qdel(existing_effect)
+
+	return apply_status_effect(
+		/datum/status_effect/stun_absorption,
+		source,
+		duration,
+		priority,
+		message,
+		self_message,
+		examine_message,
+		max_seconds_of_stuns_blocked,
+	)
+
+/**
+ * Removes all stub absorptions with the passed source.
+ *
+ * Returns TRUE if an effect was deleted, FALSE otherwise
+ */
+/mob/living/proc/remove_stun_absorption(source)
+	. = FALSE
+	for(var/datum/status_effect/stun_absorption/effect in status_effects)
+		if(effect.source != source)
+			continue
+
+		qdel(effect)
+		. = TRUE
+
+	return .
```

Comment:
I feel like this is a trivial use of the dot operator but I don't immediately know of a better way to fix it. probably no change if you can't either.